Tests/Functional/Asynchronous/Handler/main.swift

```diff
@@ -41,15 +41,40 @@ class HandlerTestCase: XCTestCase {
         XCTAssertTrue(handlerWasCalled)
     }
 
+// CHECK: Test Case 'HandlerTestCase.test_notificationNameIsObserved_handlerReturnsFalse_andFails' started.
+// CHECK: .*/Tests/Functional/Asynchronous/Handler/main.swift:53: error: HandlerTestCase.test_notificationNameIsObserved_handlerReturnsFalse_andFails : Asynchronous wait failed - Exceeded timeout of 0.1 seconds, with unfulfilled expectations: Expect notification 'returnFalse' from any object
+// CHECK: Test Case 'HandlerTestCase.test_notificationNameIsObserved_handlerReturnsFalse_andFails' failed \(\d+\.\d+ seconds\).
+    func test_notificationNameIsObserved_handlerReturnsFalse_andFails() {
+        let _ = self.expectationForNotification("returnFalse", object: nil, handler: {
+            notification in
+            return false
+        })
+        NSNotificationCenter.defaultCenter().postNotificationName("returnFalse", object: nil)
+        waitForExpectationsWithTimeout(0.1, handler: nil)
+    }
+    
+// CHECK: Test Case 'HandlerTestCase.test_notificationNameIsObserved_handlerReturnsTrue_andPasses' started.
+// CHECK: Test Case 'HandlerTestCase.test_notificationNameIsObserved_handlerReturnsTrue_andPasses' passed \(\d+\.\d+ seconds\).
+    func test_notificationNameIsObserved_handlerReturnsTrue_andPasses() {
+        let _ = self.expectationForNotification("returnTrue", object: nil, handler: {
+            notification in
+            return true
+        })
+        NSNotificationCenter.defaultCenter().postNotificationName("returnTrue", object: nil)
+        waitForExpectationsWithTimeout(0.1, handler: nil)
+    }
+
     static var allTests: [(String, HandlerTestCase -> () throws -> Void)] {
         return [
             ("test_whenExpectationsAreNotFulfilled_handlerCalled_andFails", test_whenExpectationsAreNotFulfilled_handlerCalled_andFails),
             ("test_whenExpectationsAreFulfilled_handlerCalled_andPasses", test_whenExpectationsAreFulfilled_handlerCalled_andPasses),
+            ("test_notificationNameIsObserved_handlerReturnsFalse_andFails", test_notificationNameIsObserved_handlerReturnsFalse_andFails),
+            ("test_notificationNameIsObserved_handlerReturnsTrue_andPasses", test_notificationNameIsObserved_handlerReturnsTrue_andPasses),
```

Comment:
nit-pick: Same as above, I'd prefer this in its own file, especially since the notification expectation "handler" being tested in these new tests is different from the "handler" being tested in the existing tests. I'd prefer this moved to `Tests/Functional/Asynchronous/Notifications/Handler/main.swift`